bin/run.sh

```diff
@@ -77,15 +79,54 @@ run_tests() {
     # test scripts may be (old) xxxx_test.sh
     [[ -f "$test_file" ]] || test_file="${slug//-/_}_test.sh"
 
-    sed -i 's/load bats-extra.bash/load bats-extra/' "$test_file" || true
+    perl -i -pe 's/(load bats-extra)\.bash/$1/' "$test_file"
 
     echo "Test output:"
 
     bats --tap "$test_file" 2>&1 | tee "$output_file" || true
 
+    get_test_bodies "$test_file"
+
     echo "Test run ended. Output saved in $output_file"
 
-    cd ~-
+    cd -
+}
+
+get_test_bodies() {
+    local test_file=$1
+    local name line
+    local state="out"
+    local body=() 
+    test_bodies=()
+
+    local start_test_re='^@test ['\''"](.+)['\''"] \{[[:blank:]]*$'
+    local end_test_re='^\}[[:blank:]]*$'
+
+    while IFS= read -r line; do
+        case $state in
```

Comment:
Always quote all variable expansions.

Reply:
word splitting and glob expansion don't happen for the case var, but will do.

Reply:
It's standard "good style" advice. We should practice what we preach.